Sage/carbon
# chore(check_rfcs): update to give warning on failure instead of throwing error

closes #5917

### Proposed behaviour

Update `check_rfcs` script so:
- Content is more recognisable in the terminal
- Outputs a warning in the terminal upon failure to retrieve data on open RFCs. This will prevent consumers from receiving an http error which exits the installation script. 

<img width="691" alt="Screenshot 2023-03-22 at 15 28 56" src="https://user-images.githubusercontent.com/18368713/226956274-e4296c62-1413-4b9a-86f6-8dc944916a34.png">
<img width="1441" alt="Screenshot 2023-03-22 at 15 29 03" src="https://user-images.githubusercontent.com/18368713/226956288-b864f541-a566-4e93-b3b4-5285af83676c.png">

### Current behaviour

If `check_rfcs` script fails to retrieve RFCs, any raised http error will not be caught and handled properly - causing the installation script to exit. A commonly encountered http error is when the GitHub API rate limit is exceeded:

```shell
RequestError [HttpError]: API rate limit exceeded for 51.103.107.49. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
#18 78.84     at /xtrem/app/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:86:21
#18 78.84     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
#18 78.84     at async getOpenRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:20:20)
#18 78.84     at async checkRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:42:20) {
#18 78.84   status: 403,
#18 78.84   response: {
#18 78.84     url: '[https://api.github.com/repos/sage/carbon/pulls'](https://api.github.com/repos/sage/carbon/pulls%27),
#18 78.84     status: 403,
#18 78.84     headers: {
```

### Checklist




#### QA


### Additional context



### Testing instructions



Run `npm run postinstall` to check that the script is called correctly without raising an error.
